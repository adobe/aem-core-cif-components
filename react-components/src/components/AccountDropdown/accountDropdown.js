/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import ReactDOM from 'react-dom';
import { useTranslation } from 'react-i18next';

import { useUserContext } from '../../context/UserContext';
import { useSignin } from '../SignIn/useSignin';
import LoadingIndicator from '../LoadingIndicator';
import SignOutLink from '../MyAccount/signOutLink';
import SignInForm from '../SignIn/signInForm';

import classes from './accountDropdown.css';

const AccountDropdown = () => {
    const container = document.querySelector('#account');
    const [{ isAccountDropdownOpen }] = useUserContext();
    const { errorMessage, isSignedIn, handleSubmit, inProgress } = useSignin();

    const [t] = useTranslation('account');

    const dropdownClassName = isAccountDropdownOpen ? classes.dropdown_open : classes.dropdown;
    let dropdownContent = isSignedIn ? (
        <SignOutLink />
    ) : (
        <SignInForm errorMessage={errorMessage} handleSubmit={handleSubmit} />
    );

    if (inProgress) {
        dropdownContent = <LoadingIndicator>{t('account:signing-in', 'Signing In')}</LoadingIndicator>;
    }

    return ReactDOM.createPortal(<div className={dropdownClassName}>{dropdownContent}</div>, container);
};

export default AccountDropdown;
