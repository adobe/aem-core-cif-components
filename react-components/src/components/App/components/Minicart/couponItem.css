.root {
    display: grid;
    grid-gap: 0 1rem;
    grid-template-columns: 1fr min-content;
    margin: 0 1.5rem;
}
.couponName {
    line-height: 1.5rem;
}
