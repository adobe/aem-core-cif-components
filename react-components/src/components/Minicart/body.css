.root {
    display: grid;
    grid-template-rows: min-content min-content 1fr;
    overflow: auto;
}
