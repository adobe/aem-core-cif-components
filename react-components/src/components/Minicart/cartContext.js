/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React, { createContext, useContext, useReducer } from 'react';
import { object, func } from 'prop-types';
import { useCookieValue } from '../../utils/hooks';

export const initialState = {
    isOpen: false,
    isRegistered: false,
    isEditing: false,
    isLoading: false,
    editItem: {},
    cartId: null,
    cart: null,
    errorMessage: null,
    couponError: null
};

export const reducerFactory = setCartCookie => {
    return (state, action) => {
        switch (action.type) {
            case 'close':
                return {
                    ...state,
                    isOpen: false
                };
            case 'open':
                return {
                    ...state,
                    isOpen: true
                };
            case 'beginLoading':
                return {
                    ...state,
                    isLoading: true
                };
            case 'endLoading':
                return {
                    ...state,
                    isLoading: false
                };
            case 'beginEditing':
                return {
                    ...state,
                    isEditing: true,
                    editItem: action.item
                };
            case 'endEditing':
                return {
                    ...state,
                    isEditing: false,
                    editItem: {}
                };
            case 'cartId':
                return {
                    ...state,
                    cartId: action.cartId
                };
            case 'reset':
                setCartCookie('', 0);
                return {
                    ...state,
                    cartId: null,
                    errorMessage: null,
                    couponError: null,
                    cart: null
                };
            case 'cart':
                return {
                    ...state,
                    cart: action.cart,
                    isLoading: false,
                    couponError: !action.cart.applied_coupon ? state.couponError : null
                };
            case 'register': {
                return {
                    ...state,
                    isRegistered: true
                };
            }
            case 'error':
                console.error(action.error);
                return {
                    ...state,
                    errorMessage: action.error
                };
            case 'discardError':
                return {
                    ...state,
                    errorMessage: null
                };
            case 'couponError':
                return {
                    ...state,
                    couponError: action.error
                };

            default:
                return state;
        }
    };
};

export const CartContext = createContext();

export const CartProvider = props => {
    const factory = props.reducerFactory || reducerFactory;
    const state = props.initialState || initialState;

    const [, setCartCookie] = useCookieValue('cif.cart');
    const contextValue = useReducer(factory(setCartCookie), state);

    return <CartContext.Provider value={contextValue}>{props.children}</CartContext.Provider>;
};

CartProvider.propTypes = {
    reducerFactory: func,
    initialState: object
};

export const useCartState = () => useContext(CartContext);
