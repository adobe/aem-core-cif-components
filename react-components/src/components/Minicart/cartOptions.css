/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
.root {
    display: grid;
    grid-template-rows: min-content 1fr;
    height: calc(100% - var(--minicart-header-height));
    left: 0;
    position: absolute;
    top: 3.5rem;
    width: 100%;
}

.focusItem {
    box-shadow: 0 1px rgb(var(--venia-border));
    display: grid;
    gap: 2rem;
    grid-auto-flow: column;
    grid-template-columns: 1fr;
    padding: 1rem;
}

.name {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.form {
    overflow: auto;
}

.modal {
    visibility: hidden;
    height: 100%;
    width: 100%;
    background-color: rgb(var(--venia-grey));
    text-align: center;
    position: absolute;
    bottom: 0;
}

.modal_active {
    visibility: visible;
    composes: modal;
    opacity: 0.9;
}

.options {
    font-weight: bold;
}

.quantity {
    border-color: rgb(var(--venia-border));
    border-style: solid;
    border-width: 0 0 1px;
    margin: 0 1.5rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
}

.quantityTitle {
    font-size: 1rem;
    font-weight: 600;
    line-height: 1.5;
    margin-bottom: 1.5rem;
}

.save {
    align-items: center;
    background-color: white;
    display: grid;
    grid-auto-columns: min-content;
    grid-auto-flow: column;
    grid-gap: 0.75rem;
    height: 5rem;
    justify-content: center;
    justify-items: center;
    position: relative;
}
