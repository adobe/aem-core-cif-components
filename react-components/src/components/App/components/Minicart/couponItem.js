/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useCartState } from './cartContext';
import Kebab from './kebab';
import Section from './section';

import classes from './couponItem.css';

const CouponItem = () => {
    const [{ cart, removeCoupon }] = useCartState();

    const appliedCoupon = cart.applied_coupon ? cart.applied_coupon.code : null;

    return (
        <div className={classes.root}>
            <div className={classes.couponName}>
                Coupon <strong>{appliedCoupon}</strong> applied.
            </div>
            <Kebab>
                <Section text="Remove coupon" onClick={removeCoupon} icon="Trash" />
            </Kebab>
        </div>
    );
};

export default CouponItem;
