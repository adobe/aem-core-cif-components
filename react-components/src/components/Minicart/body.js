/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { useTranslation } from 'react-i18next';

import LoadingIndicator from '../LoadingIndicator';

import EmptyMinicartBody from './emptyMinicartBody';
import classes from './body.css';
import ProductList from './productList';
import CartOptions from './cartOptions';
import Error from './error';
import CouponForm from './couponForm';
import CouponItem from './couponItem';

import { useCartState } from './cartContext';
import DiscountList from './discountList';

const CartLoadingIndicator = () => {
    const [t] = useTranslation('cart');

    return <LoadingIndicator>{t('cart:fetching-data', 'Fetching cart data...')}</LoadingIndicator>;
};

const Body = () => {
    const [{ isEditing, cart, isLoading, errorMessage }] = useCartState();
    const isEmpty = cart && Object.entries(cart).length > 0 ? cart.items.length === 0 : true;

    if (isLoading) {
        return <CartLoadingIndicator />;
    }

    if (errorMessage) {
        return <Error />;
    }

    if (isEmpty) {
        return <EmptyMinicartBody />;
    }

    if (isEditing) {
        return <CartOptions />;
    }

    const cartItems = cart.items;
    const couponFragment = !cart.applied_coupon ? <CouponForm /> : <CouponItem />;
    const discountFragment =
        cart.prices.discounts && cart.prices.discounts.length > 0 ? (
            <DiscountList discounts={cart.prices.discounts} />
        ) : (
            <div />
        );

    return (
        <div className={classes.root}>
            <ProductList cartItems={cartItems} />
            {discountFragment}
            {couponFragment}
        </div>
    );
};

export default Body;
