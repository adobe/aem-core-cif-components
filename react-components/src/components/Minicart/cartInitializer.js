/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import { useEffect } from 'react';
import { useCartState } from './cartContext';
import { useCookieValue } from '../../utils/hooks';
import { useUserContext } from '../../context/UserContext';

const CartInitializer = props => {
    const [{ cartId: stateCartId }, dispatch] = useCartState();
    const [{ cartId: registeredCartId }] = useUserContext();

    const CART_COOKIE = 'cif.cart';

    const [cartId, setCartCookie] = useCookieValue(CART_COOKIE);

    useEffect(() => {
        if (cartId && cartId.length > 0 && !stateCartId) {
            dispatch({ type: 'cartId', cartId });
        }
    }, [cartId]);

    useEffect(() => {
        if (stateCartId && (!cartId || cartId.length === 0)) {
            setCartCookie(stateCartId);
        }
    }, [stateCartId]);

    useEffect(() => {
        if (registeredCartId) {
            setCartCookie(registeredCartId);
            dispatch({
                type: 'cartId',
                cartId: registeredCartId
            });
        }
    }, [registeredCartId]);

    return props.children;
};

export default CartInitializer;
