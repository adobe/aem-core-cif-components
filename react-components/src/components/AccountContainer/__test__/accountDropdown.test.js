/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from 'test-utils';
import { CartProvider } from '../../Minicart';
import AccountDropdown from '../accountDropdown';

const config = {
    graphqlEndpoint: 'endpoint',
    storeView: 'default',
    pagePaths: {
        accountDetails: '/accountDetails'
    }
};

describe('<AccountDropdown>', () => {
    it('renders the component when account dropdown is open', () => {
        const stateWithAccountDropdownOpen = { isAccountDropdownOpen: true };
        const accountDropdownOpenClass = 'root_open';

        const { getByLabelText } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithAccountDropdownOpen }
        );
        expect(getByLabelText('account dropdown').getAttribute('class')).toEqual(accountDropdownOpenClass);
    });

    it('renders the my account component inside account dropdown', () => {
        const stateWithMyAccountView = {
            currentUser: {
                firstname: '',
                lastname: '',
                email: ''
            },
            isSignedIn: true,
            accountDropdownView: 'MY_ACCOUNT'
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithMyAccountView }
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the change password component inside account dropdown', () => {
        const stateWithChangePasswordView = {
            isSignedIn: true,
            accountDropdownView: 'CHANGE_PASSWORD'
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithChangePasswordView }
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the forgot password component inside account dropdown', () => {
        const stateWithForgotPasswordView = {
            isSignedIn: false,
            accountDropdownView: 'FORGOT_PASSWORD'
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithForgotPasswordView }
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the create account component inside account dropdown', () => {
        const stateWithCreateAccountView = {
            accountDropdownView: 'CREATE_ACCOUNT'
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithCreateAccountView }
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the account created component inside account dropdown', () => {
        const stateWithAccountCreatedView = {
            accountDropdownView: 'ACCOUNT_CREATED'
        };

        const { asFragment } = render(
            <CartProvider initialState={{ cartId: null }} reducerFactory={() => state => state}>
                <AccountDropdown />
            </CartProvider>,
            { config: config, userContext: stateWithAccountCreatedView }
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
