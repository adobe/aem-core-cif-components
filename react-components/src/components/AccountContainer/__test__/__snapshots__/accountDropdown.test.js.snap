// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDropdown> renders the account created component inside account dropdown 1`] = `
<DocumentFragment>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <div
      class="root"
    >
      <div
        class="body"
      >
        <h2
          class="header"
        >
          Your account was successfully created
        </h2>
        <div
          class="textBlock"
        >
          You will receive a link at 
          <b />
          . Access that link to confirm your email address.
        </div>
        <div
          class="actions"
        >
          <button
            class="root"
            type="button"
          >
            <span
              class="signin"
            >
              Sign In
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AccountDropdown> renders the change password component inside account dropdown 1`] = `
<DocumentFragment>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <div
      class="root"
    >
      <form>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            Current Password
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="old-password"
                class="input"
                name="oldPassword"
                type="password"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            New Password
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="password"
                autocomplete="password"
                class="input"
                name="password"
                type="password"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            Confirm New Password
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="confirm"
                class="input"
                name="confirm"
                type="password"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div
          class="actions"
        >
          <button
            aria-label="submit"
            class="root_highPriority"
            type="submit"
          >
            <span
              class="content"
            >
              Change Password
            </span>
          </button>
          <button
            aria-label="cancel"
            class="root_normalPriority"
            type="button"
          >
            <span
              class="content"
            >
              Cancel
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AccountDropdown> renders the create account component inside account dropdown 1`] = `
<DocumentFragment>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <form>
      <p>
        Check out faster, use multiple addresses, track orders and more by creating an account!
      </p>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          First Name
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="firstname"
              autocomplete="given-name"
              class="input"
              name="customer.firstname"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Last Name
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="lastname"
              autocomplete="family-name"
              class="input"
              name="customer.lastname"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Email address
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="email"
              autocomplete="email"
              class="input"
              name="customer.email"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="password"
              autocomplete="new-password"
              class="input"
              name="password"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div
        class="root"
      >
        <span
          class="label"
        >
          <span
            class="requiredSymbol"
          />
          Confirm Password
        </span>
        <span
          class="root"
        >
          <span
            class="input"
          >
            <input
              aria-label="confirm"
              class="input"
              name="confirm"
              type="password"
              value=""
            />
          </span>
          <span
            class="before"
          />
          <span
            class="after"
          />
        </span>
        <p
          class="root"
        />
      </div>
      <div>
        <label
          class="root"
        >
          <span
            class="icon"
          />
          <input
            aria-label="subscribe"
            class="input"
            name="subscribe"
            type="checkbox"
          />
          <span
            class="label"
          >
            Subscribe to news and updates
          </span>
        </label>
        <p
          class="root"
        />
      </div>
      <div />
      <div>
        <button
          aria-label="submit"
          class="root_highPriority"
          type="submit"
        >
          <span
            class="content"
          >
            Submit
          </span>
        </button>
        <button
          aria-label="cancel"
          class="root_normalPriority"
          type="button"
        >
          <span
            class="content"
          >
            Cancel
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`<AccountDropdown> renders the forgot password component inside account dropdown 1`] = `
<DocumentFragment>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <div
      class="root"
    >
      <p
        class="instructions"
      >
        Enter your email below to receive a password reset link.
      </p>
      <form>
        <div
          class="root"
        >
          <span
            class="label"
          >
            <span
              class="requiredSymbol"
            />
            Email address
          </span>
          <span
            class="root"
          >
            <span
              class="input"
            >
              <input
                aria-label="email"
                autocomplete="email"
                class="input"
                name="email"
                value=""
              />
            </span>
            <span
              class="before"
            />
            <span
              class="after"
            />
          </span>
          <p
            class="root"
          />
        </div>
        <div>
          <button
            aria-label="submit"
            class="root_highPriority"
            type="submit"
          >
            <span
              class="content"
            >
              Submit
            </span>
          </button>
          <button
            aria-label="cancel"
            class="root_normalPriority"
            type="button"
          >
            <span
              class="content"
            >
              Cancel
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<AccountDropdown> renders the my account component inside account dropdown 1`] = `
<DocumentFragment>
  <button
    data-role="mask"
  />
  <div
    aria-label="account dropdown"
    class="root"
  >
    <div
      class="root"
    >
      <div
        class="user"
      >
        <h2
          class="title"
        >
           
        </h2>
        <span
          class="subtitle"
        />
      </div>
      <div
        class="actions"
      >
        <button
          classes="[object Object]"
          type="button"
        >
          <span>
            <span>
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="11"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="11"
                />
                <path
                  d="M7 11V7a5 5 0 0 1 10 0v4"
                />
              </svg>
            </span>
            <span>
              Change Password
            </span>
          </span>
        </button>
        <button
          classes="[object Object]"
          type="button"
        >
          <span>
            <span>
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 19.5A2.5 2.5 0 0 1 6.5 17H20"
                />
                <path
                  d="M6.5 2H20v20H6.5A2.5 2.5 0 0 1 4 19.5v-15A2.5 2.5 0 0 1 6.5 2z"
                />
              </svg>
            </span>
            <span>
              Address Book
            </span>
          </span>
        </button>
        <button
          classes="[object Object]"
          type="button"
        >
          <span>
            <span>
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="16"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12.01"
                  y1="8"
                  y2="8"
                />
              </svg>
            </span>
            <span>
              Account Information
            </span>
          </span>
        </button>
        <button
          classes="[object Object]"
          type="button"
        >
          <span>
            <span>
              <svg
                fill="none"
                height="18"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
                />
                <polyline
                  points="16 17 21 12 16 7"
                />
                <line
                  x1="21"
                  x2="9"
                  y1="12"
                  y2="12"
                />
              </svg>
            </span>
            <span>
              Sign Out
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
